IllegalStateException  after ExternalPlayerFragment was destroyed

**App version:** 1.6.6 RC

**Android version**: all

**Device model**: all

**Stacktrace/Logcat**: 
```
java.lang.IllegalStateException: 
  at rx.android.schedulers.LooperScheduler$ScheduledAction.run (LooperScheduler.java:114)
  at android.os.Handler.handleCallback (Handler.java:790)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:171)
  at android.app.ActivityThread.main (ActivityThread.java:6633)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:547)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:823)
Caused by: java.lang.NullPointerException: 
  at com.bumptech.glide.manager.RequestManagerRetriever.assertNotDestroyed (RequestManagerRetriever.java:133)
  at com.bumptech.glide.manager.RequestManagerRetriever.get (RequestManagerRetriever.java:102)
  at com.bumptech.glide.Glide.with (Glide.java:681)
  at de.danoeh.antennapod.fragment.ExternalPlayerFragment$$Lambda$4.call (Unknown Source:1193)
  at rx.Single$11.onSuccess (Single.java:1803)
  at rx.internal.operators.SingleObserveOn$ObserveOnSingleSubscriber.call (SingleObserveOn.java:88)
  at rx.android.schedulers.LooperScheduler$ScheduledAction.run (LooperScheduler.java:107)
```

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -150,7 +150,7 @@ dependencies {
     implementation "com.android.support:gridlayout-v7:$supportVersion"
     implementation "com.android.support:percent:$supportVersion"
     implementation "com.android.support:recyclerview-v7:$supportVersion"
-    provided 'com.google.android.wearable:wearable:2.2.0'
+    compileOnly 'com.google.android.wearable:wearable:2.2.0'
     implementation "org.apache.commons:commons-lang3:$commonslangVersion"
     implementation("org.shredzone.flattr4j:flattr4j-core:$flattr4jVersion") {
         exclude group: "org.json", module: "json"

--- a/app/src/main/java/de/danoeh/antennapod/fragment/ExternalPlayerFragment.java
+++ b/app/src/main/java/de/danoeh/antennapod/fragment/ExternalPlayerFragment.java
@@ -20,10 +20,10 @@ import de.danoeh.antennapod.R;
 import de.danoeh.antennapod.core.feed.MediaType;
 import de.danoeh.antennapod.core.glide.ApGlideSettings;
 import de.danoeh.antennapod.core.service.playback.PlaybackService;
-import de.danoeh.antennapod.core.util.Converter;
 import de.danoeh.antennapod.core.util.playback.Playable;
 import de.danoeh.antennapod.core.util.playback.PlaybackController;
 import rx.Single;
+import rx.Subscription;
 import rx.android.schedulers.AndroidSchedulers;
 import rx.schedulers.Schedulers;
 
@@ -41,6 +41,7 @@ public class ExternalPlayerFragment extends Fragment {
     private TextView mFeedName;
     private ProgressBar mProgressBar;
     private PlaybackController controller;
+    private Subscription subscription;
 
     public ExternalPlayerFragment() {
         super();
@@ -81,7 +82,7 @@ public class ExternalPlayerFragment extends Fragment {
         super.onActivityCreated(savedInstanceState);
         controller = setupPlaybackController();
         butPlay.setOnClickListener(v -> {
-            if(controller != null) {
+            if (controller != null) {
                 controller.playPause();
             }
         });
@@ -142,6 +143,9 @@ public class ExternalPlayerFragment extends Fragment {
         if (controller != null) {
             controller.release();
         }
+        if (subscription != null) {
+            subscription.unsubscribe();
+        }
     }
 
     @Override
@@ -162,7 +166,7 @@ public class ExternalPlayerFragment extends Fragment {
         controller = setupPlaybackController();
         if (butPlay != null) {
             butPlay.setOnClickListener(v -> {
-                if(controller != null) {
+                if (controller != null) {
                     controller.playPause();
                 }
             });
@@ -177,7 +181,10 @@ public class ExternalPlayerFragment extends Fragment {
             return false;
         }
 
-        Single.create(subscriber -> subscriber.onSuccess(controller.getMedia()))
+        if (subscription != null) {
+            subscription.unsubscribe();
+        }
+        subscription = Single.create(subscriber -> subscriber.onSuccess(controller.getMedia()))
                 .subscribeOn(Schedulers.newThread())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(media -> updateUi((Playable) media));
@@ -206,13 +213,8 @@ public class ExternalPlayerFragment extends Fragment {
                 butPlay.setVisibility(View.VISIBLE);
             }
         } else {
-            Log.w(TAG,  "loadMediaInfo was called while the media object of playbackService was null!");
+            Log.w(TAG, "loadMediaInfo was called while the media object of playbackService was null!");
         }
-    }
-
-    private String getPositionString(int position, int duration) {
-        return Converter.getDurationStringLong(position) + " / "
-                + Converter.getDurationStringLong(duration);
     }
 
     public PlaybackController getPlaybackControllerTestingOnly() {

